src/omnisharpServer.ts

```diff
@@ -57,9 +135,30 @@ export abstract class OmnisharpServer {
 	private _setState(value: ServerState) : void {
 		if (typeof value !== 'undefined' && value !== this._state) {
 			this._state = value;
-			this._fireEvent('stateChanged', this._state);
+			this._fireEvent(Events.StateChanged, this._state);
 		}
 	}
+    
+    private _recordDelay(path: string, elapsedTime: number) {
+        let delays = this._telemetryDelays[path];
+        if (!delays) {
+            delays = new Delays();
+            this._telemetryDelays[path] = delays;
+        }
+        
+        delays.report(elapsedTime);
+    }
+    
+    public reportAndClearTelemetry() {
+        for (var path in this._telemetryDelays) {
+            const eventName = 'omnisharp' + path;
```

Comment:
no it doesn't. It's the "path" to the request. You're right. I'll pick a better name here.